java/src/main/java/genericpubsub/ManagedSubscribe.java

```diff
@@ -153,32 +165,34 @@ public void onNext(ManagedFetchResponse fetchResponse) {
         }
 
         synchronized (this) {
-            receivedEvents += batchSize;
+            receivedEvents.addAndGet(batchSize);
             this.notifyAll();
-            if (!isActive) {
+            if (!isActive.get()) {
                 return;
             }
         }
+
         if (fetchResponse.getPendingNumRequested() == 0) {
             fetchMore(BATCH_SIZE);
         }
     }
 
     @Override
     public void onError(Throwable throwable) {
+        printStatusRuntimeException("Error during subscribe stream", (Exception) throwable);
```

Comment:
Moving this higher up to ensure that the logs are printed before things shutdown. Without this change the exception wasn't getting logged when I ran things locally